torch/_dynamo/guards.py

```diff
@@ -643,19 +643,9 @@ def compile_check_fn(
         )
         for guard in aotautograd_guards:
             if isinstance(guard, DuplicateInputs):
-                pos_a = self.output_graph.pos_to_arg[guard.input_pos_a]
-                pos_b = self.output_graph.pos_to_arg[guard.input_pos_b]
-                assert (
-                    pos_b >= 0 and pos_a >= 0
-                ), "Deduped args out of bounds, cannot be negative"
-
-                assert self.output_graph.graphargs[
-                    pos_a
-                ].is_tensor, "Deduped arg must be a tensor"
-                assert self.output_graph.graphargs[
-                    pos_b
-                ].is_tensor, "Deduped arg must be a tensor"
-                code_part = f"{self.output_graph.graphargs[pos_a].source.name()} is {self.output_graph.graphargs[pos_b].source.name()}"  # noqa: B950
+                source_a = guard.input_source_a
+                source_b = guard.input_source_b
+                code_part = f"{source_a.name()} is {source_b.name()}"
```

Comment:
i feel like it'd be good to be explicit about matching the logic used here (a is b) with the logic used in aot when generating the dupe args map in the first place.

a way to do that nicely might be to factor it out as a function `is_duplicate(a, b)` which we call in both places.

a related question is what is the correct implementation for `is_duplicate` - should it care about pyobj or c++ tensorImpl or some alias property or what?

(i think we can probably land this as is first and follow up, since it seems strictly better than status quo)